### PR TITLE
Psalm 1

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Storage/FileStorage.php [hunk 1]
```diff
@@ -49,16 +48,17 @@ public function setHistorySize(int $historySize): void
...
     {
         clearstatcache();
-        $data = [];
-        $pattern = sprintf(
-            '%s/**/%s/%s.json',
-            $this->path,
-            $id ?? '**',
-            $type,
+
+        $dataFiles = $this->findFilesOrderByModifiedTime(
```
Comment (samdark) [inline src/Storage/FileStorage.php:52]:
```suggestion
        $dataFiles = $this->findFilesOrderedByModifiedTime(
```

Comment (vjik) [reply src/Storage/FileStorage.php]: done

File: src/Collector/ContainerInterfaceProxy.php [hunk 2]
```diff
@@ -163,10 +176,11 @@ private function getServiceProxyFromArray(object $instance, array $params): ?obj
...
                     } catch (Exception) {
-                        //leave as is
+                        // leave as is
                     }
                 }
             }
+            /** @psalm-suppress MixedMethodCall */
             return new $proxyClass($instance, ...$params);
         } catch (Exception) {
             return null;
```
Comment (samdark) [inline src/Collector/ContainerInterfaceProxy.php:186]: Does that work with `ObjectProxy` return type?

**Review by samdark — APPROVED** [review]

File: src/Collector/ContainerInterfaceProxy.php (hunk 2 above)
Comment (vjik) [reply src/Collector/ContainerInterfaceProxy.php:186]: This method returns `?object`

File: src/Collector/Stream/HttpStreamCollector.php [hunk 3]
```diff
@@ -76,9 +90,7 @@ public function getSummary(): array
             'http_stream' => array_merge(
                 ...array_map(
                     fn (string $operation) => [
-                        $operation => is_countable($this->requests[$operation]) ? count(
-                            $this->requests[$operation]
-                        ) : 0,
+                        $operation => count($this->requests[$operation]),
```
Comment (xepozz) [inline src/Collector/Stream/HttpStreamCollector.php:93]: I don't remember why, but it was not just for fun

File: src/ProxyDecoratedCalls.php [hunk 4]
```diff
@@ -4,9 +4,6 @@
 
 namespace Yiisoft\Yii\Debug;
 
-/**
- * @property object $decorated
```
Comment (xepozz) [inline src/ProxyDecoratedCalls.php:8]: Why?

File: src/Storage/FileStorage.php [hunk 5]
```diff
@@ -143,4 +142,21 @@ private function gc(): void
...
+     * @return string[]
+     */
+    private function findFilesOrderedByModifiedTime(string $pattern): array
+    {
+        $files = glob($pattern, GLOB_NOSORT);
+        if ($files === false) {
+            return [];
+        }
+
+        usort(
```
Comment (xepozz) [inline src/Storage/FileStorage.php:156]: Is it ok it's changed from `uasort`? Looks like the result keys aren't used here, but anyway

**Review by xepozz — CHANGES_REQUESTED** [review]

File: src/Collector/Stream/HttpStreamCollector.php (hunk 3 above)
Comment (vjik) [reply src/Collector/Stream/HttpStreamCollector.php:93]: `requests` property always contains array of lists (type `array<string, list<array{uri: string, args: array}>>`)
So countable check is not need.

File: src/ProxyDecoratedCalls.php (hunk 4 above)
Comment (vjik) [reply src/ProxyDecoratedCalls.php:8]: This causes many psalm errors in classes used trait.

File: src/Storage/FileStorage.php (hunk 5 above)
Comment (vjik) [reply src/Storage/FileStorage.php:156]: Yes, keys never used.

**Review by samdark — APPROVED** [review]

File: src/Collector/Stream/HttpStreamCollector.php (hunk 3 above)
Comment (xepozz) [reply src/Collector/Stream/HttpStreamCollector.php:93]: Maybe it was originally in the http stream proxy and so the requests were overridden in php engine, but anyway let's pass it now, will check it later

**Review by xepozz — APPROVED** [review]

**Pull request merged**